### PR TITLE
Create FFI plugin

### PR DESCRIPTION
**Pull request by mahesh-hegde**
Commit history seems neat now.

I think it's better to add `build_example` workflow in commit adding android example because in my repo, proper example is added after android support. In next commit I can add `dart test`.

Files changed after flutter create: android/build.gradle (to bump NDK version)

### CODE REVIEW

**Comment by dcharkes** [comment]
> Commit history seems neat now.

👍  nice

> I think it's better to add `build_example` workflow in commit adding android example because in my repo, proper example is added after android support.

I think it's better to have it now, then we at least test that all the Flutter boilerplate works. In general, it's always better to have code covered by automated tests than to have code not covered by automated tests.

This will also mean that when you add Android support, the workflow will automatically test if the build for Android works.
Similarly, when we add the native code to be build with Macos/Windows/Linux, the workflow will test that that build succeeds.

> In next commit I can add `dart test`.

👍 

> Files changed after flutter create: android/build.gradle (to bump NDK version)

Super, that's easy to review 😄 

I think with the workflow testing the Flutter build we can merge this.

**Review by dcharkes — APPROVED** [review]
lgtm!

File: .github/workflows/test-package.yml [hunk 1]
```diff
@@ -73,3 +73,68 @@ jobs:
...
+    defaults:
+      run:
+        working-directory: ./jni/example
+    steps:
+      - uses: actions/checkout@v3
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+      - run: |
+         sudo apt-get update -y
```
[the remark this answers is not in the document]
Comment (mahesh-hegde) [reply .github/workflows/test-package.yml:88]: Not sure, I copied from the flutter action's example.

**Comment by dcharkes** [comment]
Thanks @mahesh-hegde!  🚀

**Pull request merged**